src/lib/oauth2-server.ts

```diff
@@ -37,12 +39,25 @@ export class OAuth2Server extends HttpServer {
 
   /**
    * Creates a new instance of OAuth2Server.
+   *
+   * @param {string | undefined} key Optional key file path for ssl
+   * @param {string | undefined} cert Optional cert file path for ssl
    */
-  constructor() {
+  constructor(key?: string, cert?: string) {
     const iss = new OAuth2Issuer();
     const serv = new OAuth2Service(iss);
 
-    super(serv.requestHandler);
+    let options: ServerOptions | undefined = undefined;
+    if (key && cert) {
```

Comment:
@lbestftr When only one of the two params is set, this will silently default to creating the server as http. Could you rather make this throw?

Reply:
Fixed!